### PR TITLE
[Filebeat] Add HMAC template function for httpjson input

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds a `[[hmac "sha256" "secret" "string1" (formatDate (now) "RFC1123")]]` function that can be used to calculate hmac signatures for API authentication.

## Why is it important?

A lot of APIs use an HMAC signature to authenticate API calls.  These signatures need to be generated at runtime, usually comprised of elements of the call. See https://docs.fireeye.com/iSight/index.html#/authentication as an example. Also https://elasticstack.slack.com/archives/CNEDGGJQ3/p1627286354468400

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Run the integration tests located at x-pack/filebeat/input/httpjson/internal/v2/value_tpl_test.go

## Related issues

- 

## Use cases

https://docs.fireeye.com/iSight/index.html#/authentication

### CODE REVIEW

**Comment by legoguy1000** [comment]
@kaiyan-sheng I'd be interested in your thoughts on this one.

**Review by kaiyan-sheng — COMMENTED** [review]
This looks good to me. But I'm not familiar with the httpjson input code :) @P1llus @marc-gr Could you take a look please? Thanks!

File: x-pack/filebeat/input/httpjson/internal/v2/value_tpl.go [hunk 1]
```diff
@@ -236,3 +242,26 @@ func mul(a, b int64) int64 {
 func div(a, b int64) int64 {
 	return a / b
 }
+
+func hmac_string(hmac_type string, hmac_key string, values ...string) string {
```
Comment (marc-gr) [inline x-pack/filebeat/input/httpjson/internal/v2/value_tpl.go:246]:
```suggestion
func hmacString(hmacType string, hmacKey string, values ...string) string {
```

Just for consistency with the rest.

**Review by marc-gr — COMMENTED** [review]
All looks good! just a minor renaming for consistency and will be good to go :)

**Comment by legoguy1000** [comment]
@marc-gr changes made.

**Review by marc-gr — APPROVED** [review]

**Pull request merged**